build/scripts/obtain_mono.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+# Source this script to ensure mono is installed and on the path.
+
+# This is a function to keep variable assignments out of the parent script (that is sourcing this file)
+install_mono () {
+    # Download and install `mono` locally
+    local THIS_DIR="$(cd -P "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+    source "${THIS_DIR}"/build-utils.sh
+
+    local MONO_VERSION="$(get_tool_version mono)"
+    # the tar file has `mono` as the root directory
+    local MONO_PATH="${THIS_DIR}"/../../Binaries/Tools
+
+    if [[ ! -x "${MONO_PATH}/mono/bin/mono" ]]
+    then
+        echo "Downloading mono ${MONO_VERSION}"
+        mkdir -p "${MONO_PATH}"
+        curl -L https://roslyninfra.blob.core.windows.net/jenkins/mono/mono-${MONO_VERSION}.tar.gz | tar xz -C "${MONO_PATH}"
```

Comment:
Is there a reason we're having to use a (apparently custom) version of mono vs. some official .deb? I'm assuming there *is* a good reason, but wonder where we've documented that. :smile:

Reply:
Where would you expect such documentation to exist? The version is in Tools.props (as is the case with other external tools we depend on),

Reply:
Lacking a better idea, a comment before that curl line.

Reply:
> some official .deb?

How would you install the .deb without root? (you can't)

If your answer is "unpack it", there is no single .deb - how do you get a list of the required .debs? (`mono-complete` depends on hundreds of packages)

If your answer is "download the package lists/dependency graph from mono's package servers and find the transitive closure", congrats, you've reimplemented a linux package manager. I didn't want to do that.